proc/proc.go

```diff
@@ -0,0 +1,249 @@
+// proc offers control over arbitrary processes.
+// proc depends heavily on os/exec.
+// If/when proc needs to do resource control & isolation
+// its probable that os/exec will need to be replaced with
+// either the os package itself or syscall.
+package proc
+
+// Fingers crossed that its easier than rewriting os.ForkExec
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+const outputReadRefreshInterval = time.Millisecond * 10
+
+// Proc is a BitBox process.
+// Stderr and Stdout are dumped to temp files on disk.
+type Proc struct {
+	// TODO: there is 99% chance we need a synchronization primitive at this level
+	stdoutFileName string
+	stderrFileName string
+	cmd            *exec.Cmd
+}
+
+// NewProc constructs and begins process.
+// Stdout & Stderr of the new process are pointed at temp files.
+// The tempfileNames are acessable through the coresponding members.
+func NewProc(cmdName string, args ...string) (Proc, error) {
+
+	var err error
+	var cmdPath string
+	if cmdPath, err = exec.LookPath(cmdName); err != nil {
```

Comment:
lookpath will be called by the `exec.Comman` call. So I think it's not really needed here

Reply:
I've made the same mistake, I don't think I checked the docs as it just seemed non intuitive that Exec does this.